src/auth/views/register.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { Formik } from 'formik';
+import { Link } from 'react-router-dom';
+import Input from 'common/components/input';
+import SubmitButton from 'common/components';
+import { useTranslation } from 'react-i18next';
+import { AuthLayout } from 'layouts/auth-layout';
+import { RegisterPayload } from 'auth/auth.types';
+import { registerRestaurant } from 'api/request.api';
+import { useModal, Modal } from 'common/components/modal';
+import { registerValidationSchema } from 'auth/auth.validation';
+
+const Register = () => {
+  const registerModal = useModal(true);
+  const { t } = useTranslation();
+
+  const initialValues: RegisterPayload = {
+    name: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  };
```

Comment:
Let's move Outside the Component. As mentioned above these helps just to increase the number of lines of method. We can always try to make method as small as possible.